google/cloud/pubsub_v1/publisher/client.py

```diff
@@ -369,38 +372,54 @@ def publish(self, topic, data, ordering_key="", **attrs):
                 "be sent as text strings."
             )
 
-        # Create the Pub/Sub message object.
-        message = types.PubsubMessage(
-            data=data, ordering_key=ordering_key, attributes=attrs
-        )
+        span_name = "{} publisher".format(topic)
+        span_attributes = {"data": data.decode()}
+        with create_span(span_name, attributes=span_attributes) as span:
+            if span is not None:
 
-        # Messages should go through flow control to prevent excessive
-        # queuing on the client side (depending on the settings).
-        try:
-            self._flow_controller.add(message)
-        except exceptions.FlowControlLimitError as exc:
-            future = futures.Future()
-            future.set_exception(exc)
-            return future
+                if "googclient_OpenTelemetrySpanContext" in attrs:
+                    _LOGGER.warn(
+                        "googclient_OpenTelemetrySpanContext set on message"
+                        "as an attribute, but will be overridden."
+                    )
```

Comment:
Oh, forgot about this - I don't think we test this code path yet, i.e. whether a warning is indeed emitted.